File: frost-cpp/frost-scripts/performance-test/large-map-access-workload.rb
#!/usr/bin/env ruby

n_keys = 5_000
n_lookups = 500_000

m = (0...n_keys).to_h { |i| [i.to_s, i * i] }

result = (0...n_lookups).sum { |j| m[(j % n_keys).to_s] }
raise unless result == 4_165_416_750_000
puts result

File: frost-cpp/frost-scripts/performance-test/large-map-access-workload.py
#!/usr/bin/env python3

n_keys = 5_000
n_lookups = 500_000

m = {str(i): i * i for i in range(n_keys)}

result = sum(m[str(j % n_keys)] for j in range(n_lookups))
assert result == 4_165_416_750_000
print(result)
